[WIP] Adding a section on vendor vs. proxy to the FAQ

schafer14 opened the pull request:
**What is the problem I am trying to address?**

There have been a few questions in the recent past about when/why to use a
vendor directory instead of (or along with) Athens or another proxy.

**How is the fix applied?**

I added a section to the FAQ that explains some pros and cons for when to use
vendor and when to use a proxy server like Athens.

**Mention the issue number it fixes or add the details of the changes if it doesn't have a specific issue.**

Fixes #1338

<!-- 
example: Fixes #123
-->

Hopefully this is helpful. I just added some examples from the discussion to the bullet points. I also edited some of the writing I found confusing, but I'm not sure it helped to much. I think at this stage we can delete the conversation and focus on the summation.

Commit:
Merge remote-tracking branch 'arschles/explain-vendor-vs-proxy' into origin/master

Commit:
adding to vendor vs athens faq

arschles commented:
@schafer14 did you mean to submit this PR against my fork?

schafer14 commented:
@arschles nope I don't think so.  I was trying to build on #1340 in gomods/athens, but I got a little confused. I think I probably should have asked what to do first, but got a bit of stage fright. 

I guess I want to know how to add additions to a PR in the WIP phase, or if I should just create another PR. I think I should have put a comment in the original issue saying I was going to do some work on the issue, but considering there was already a PR I wasn't certain what to do. Sorry about the confusion.

arschles commented:
All good, @schafer14 - I will fix conflicts and merge in your change

arschles commented:
No worries @schafer14 - I get what you're trying to do now. can you fix the conflict and I'll merge this?

arschles committed:
Removing gofmt vendor ignores (#1321)

* Removing gofmt vendor ignores

We don't have a vendor directory anymore :smile: :tada:

* wrapping env var in quotes

Thanks @chriscoffee for the suggestion!

Commit:
Merge remote-tracking branch 'arschles/explain-vendor-vs-proxy' into explain-vendor-vs-proxy

schafer14 commented:
@arschles thanks. I hope this is helpful.

knsh14 committed:
Fix typo (#1358)

* fix duplicated 'the'

* fix typo found by misspell

epk committed:
Allow full name override in helm chart (#1348)

* Allow full name override in helm chart

* Update Chart.yaml

Commit:
docs: Fix outdated validation hook's link (#1351)

Commit:
Remove ref to vendor for go build in e2e script. (#1360)

manugupt1 committed:
Add docs for fargate url (#1352)

arschles committed:
Adding a note in the configuration section about the config.dev.toml (#1343)

Fixes https://github.com/gomods/athens/issues/921

marpio committed:
Enable Azure storage and stasher tests in drone (#1281)

* enable azure ci tests

* add import

twexler committed:
Don't return an http error when a DownloadMode is invalid (#1365)

* Don't return an http error when a DownloadMode is invalid

Also adds some basic testing for `mode.NewFile`

Refs #1336

* Address PR comments

doddi committed:
added environment argument for golang version (#1361)

* added environment argument for golang version

* Revert: added environment argument for golang version (84298506)

* reverted to specifying go version 1.12 as the default

* fix spacing

arschles committed:
Updating the download config docs to indicate how to point to the download file (#1371)

marwan-at-work committed:
Upgrade to Go 1.13 (#1377)

arschles committed:
removing the triage issue template (#1362)

knsh14 committed:
Fix ignoring error (#1356)

* fix ignoring error

* fix operator

* fix minio error handling

* fix get url to return error if failed parsing

* add error handling

* position of statement to get redirect url was wrong.

snxq committed:
feat: add /robots.txt (#1355)

* feat: add /robots.txt

* disallow all crawler by default.

gudladona committed:
Fix helm chart to point to the latest docker image tag (#1381)

* Update Chart to 0.6.0 docker tag

* Upgrade app version to 0.6.0

* Bump version to 0.3.3

marwan-at-work committed:
proxy/actions/sumdb: copy Go behavior for pattern matching (#1388)

marwan-at-work committed:
cmd/proxy/actions: pass NoSumPatterns to GoGetFetcher (#1391)

* cmd/proxy/actions: pass NoSumPatterns to GoGetFetcher

* add gosum unit tests

* upgrade appveyor to 1.13

* fix test typo

* check appveyor version

* AppVeyor has no 1.13

* use logger

wangkirin committed:
update import path and go mod of opencensus (#1392)

gowdabas committed:
Use upstream proxy for docker unit and e2e tests. (#1394)

* Adds cataloger for Azure Blob storage

* Revert "Adds cataloger for Azure Blob storage"

This reverts commit 9476c9e2569ad71eb49bb39628c3ebc936f2d917.

* Implements Cataloger interface for Azure blob storage. Fixes #995.

* Upgraded azblob version to v0.7.0

* Upgrades azblob version to v0.7.0

* Fix formatting

* Fix formatting issues

* Use keyed field for comp literal

* reorder import pkgs

* Use upstream proxy in test containers

* delete unwanted file

* Updates GOLANG_VERSION for docker tests

wangkirin committed:
Add chinese translation of known-limitations.md  and why.md  (#1384)

* add chinese translation of known-limitations

* add chinese translation of why

Commit:
fix proxy not work when lower case is unset. (#1332)

* fix proxy not enable when proxy lower is unset.

* Update prepare_env.go

format blankspace

Commit:
Merge branch 'master' of https://github.com/gomods/athens into explain-vendor-vs-proxy

Commit:
Merge branch 'explain-vendor-vs-proxy' of github.com:schafer14/athens into explain-vendor-vs-proxy

Commit:
removing reference material from FAQ when to use Athens

kmhagan committed:
Allow EnvList.Validate to have multiple = for GODEBUG var (#1397)

* Allow EnvList.Validate to have multiple = for GODEBUG var

* Added tests for EnvList GODEBUG.

* Made EnvList.Validate less strict, just checks for existence of one equal sign

jorng committed:
Use glob matching for package paths (#1409)

- Match the behavior of GOPRIVATE/GONOPROXY/GONOSUMDB

abhirockzz committed:
Added storage configuration section for Azure Blob Storage (#1411)

RobAtticus committed:
Add ability to annotate the proxy Service via Helm (#1402)

It is often use to add custom annotations to a Service deployed
through Helm, so this PR adds the ability for a user to specify
annotations in their values file. For such use case we have seen
is when working in conjunction with external-dns, we want to assign
DNS properties of the Service, but currently cannot because this
Helm chart does not support it.

Commit:
Correct issue with continuation token failures on deploy (#1415)

* Add working image for minio

Between RELEASE.2019-09-26T19-42-35Z and RELEASE.2019-10-02T21-19-38Z
there's an issue with continuation tokens that's caused #1414 and #1413
to fail.

Signed-off-by: Chris M <me@christophermills.co.uk>

* Undo docker changes, update minio to v6

* Narrow issue down to catalog compliance test

Signed-off-by: Chris M <me@christophermills.co.uk>

arschles committed:
Merge branch 'master' into explain-vendor-vs-proxy

Commit:
Fixing awkward phrasing in faq docs.

chengjingtao committed:
Update tag of gomods/athens in chart (#1419)

Fix an import bug: https://github.com/gomods/athens/issues/1390

Signed-off-by: jtcheng <jtcheng0616@gmail.com>

arschles committed:
Merge branch 'master' into explain-vendor-vs-proxy

bndw committed:
ci: publish git commit Docker tags (#1413)

* ci: publish git commit SHA tags

Signed-off-by: bndw <benjamindwoodward@gmail.com>

* ci: publish commit sha tags to gomods/athens-dev

Signed-off-by: bndw <benjamindwoodward@gmail.com>

ernado committed:
pkg/module: set only present environment variables (#1399)

* pkg/module: set only present environment variables

* pkg/module: apply new env variable logic to all keys

arschles committed:
Merge branch 'master' into explain-vendor-vs-proxy

Commit:
Explain vendor vs proxy (#1403)

* adding a section for discussion on when to do vendor directories vs. when to use athens or another proxy server

* adding more detail and an outline

* Adding a bit more :smile:

* adding to vendor vs athens faq

* removing reference material from FAQ when to use Athens

* Fixing awkward phrasing in faq docs.

nmiyake committed:
Remove extraneous period in DEVELOPMENT.md (#1425)

Commit:
updating FAQ docs based on [slack message](https://gophers.slack.com/archives/C9LRAQN8N/p1571149854034400?thread_ts=1571093549.033500&cid=C9LRAQN8N)

Commit:
Merge remote-tracking branch 'upstream/master' into explain-vendor-vs-proxy

davidpfarrell committed:
Add .git to .dockerignore (#1429)

arschles committed:
Adding note about office hours, removing note about developer meetings (#1344)

* Adding note about office hours, removing note about developer meetings

@LinuxSuRen @lopetazor would you mind checking out the Chinese / Spanish documentation (respectively) and changing the wording in there, when you can?

* Changing title

thanks @robjloranger!

* Some updates

For @jpreese

* Making a proper link to the zoom download URL

Thanks for pointing this out @jpreese!

wangkirin committed:
Add chinese translation of install section (#1444)

* add index translation of install section

* add translation of build-from-source

* add chinese translation of install-on-aci.md

* add chinese translation of intall-on-kubernetes

* add zh translation of managing-private-repos

* fix some typos

davidpfarrell committed:
Add GOLANG_VERSION arg support to proxy-docker make target + Dockerfile (#1430)

* Add GOLANG_VERSION arg to proxy-docker

* Set default GOLANG_VERSION to 1.13 in Makefile

* Add note in proxy Dockerfile re: GOLANG_VERSION

* Add GOLANG_VERSION to docker-compose dev target

Commit:
Remove mention of Travis from deployment script (#1446)

Signed-off-by: Chris M <millscj01@gmail.com>

davidpfarrell committed:
Remove G_ENV=production from proxy Dockerfile (#1443)

marwan-at-work committed:
drone: give more time for athens to boot up (#1451)

davidpfarrell committed:
Invoke envOverrides before checkFilePerms (#1441)

* Invoke envOverrides before checkFilePerms

* Reverts changes to cmd/proxy/Dockerfile

nikohaap committed:
Added configurations for k8s livenessProbs thresholds (#1447)

* Added configurations for k8s livenessProbs thresholds

* Fixed the Helm chart version

arschles committed:
Merge branch 'master' into explain-vendor-vs-proxy
